python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -172,6 +175,37 @@ def resolve_partitionKeys(self, _graphene_info):
                 ]
         return []
 
+    def resolve_latestMaterializationPerPartition(self, _graphene_info, **kwargs):
+        from ..implementation.fetch_assets import get_asset_events
+
+        get_timestamp = lambda event: event.timestamp
+        get_partition = (
+            lambda event: event.dagster_event.step_materialization_data.materialization.partition
+        )
+
+        partitions = kwargs.get("partitions")
+        events_for_partitions = get_asset_events(
+            _graphene_info,
+            self._external_asset_node.asset_key,
+            partitions,
+        )
+
+        latest_materialization_by_partition = {}
+        for event in events_for_partitions:
+            event_partition = get_partition(event)
+            if event_partition not in latest_materialization_by_partition:
+                latest_materialization_by_partition[event_partition] = event
+            else:
```

Comment:
It we do use sorting to rm this else case, it might be nice to add a `if len(latest_materialization_by_partition) == len(partitions): break` to avoid iterating over all the older data if the result set has been filled.

Reply:
Thanks ben! I double checked and indeed the events are already ordered from newest to oldest. I followed your suggestions and added the `break`